geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/messenger/JGroupsMessenger.java

```diff
@@ -379,6 +393,10 @@ public void started() {}
 
   @Override
   public void stop() {
+    if (localAddress.getVmViewId() >= 0) {
```

Comment:
I see this block repeated in emergencyClose. Consider extracting it to its own method